### PR TITLE
Ported Bay's port of VG's holomaps.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
Ports the holomap subsystem from @CrimsonShrike's port of VG's holomaps.
Opening for review/diff, so far I've just managed to get it to compile.

## Why and what will this PR improve
Holomaps are a fun little feature that autogenerates an ingame map.

image

## Authorship
@CrimsonShrike for the port: https://github.com/Baystation12/Baystation12/pull/31833
@JMWTurner for the original implementation: https://github.com/vgstation-coders/vgstation13/pull/11749

## Changelog
:cl:
add: Ported Bay's port of VG's holomaps.
/:cl:

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Reviews/input on this welcome, I'm expecting we're going to need to do a bit of hatchet work before merge.

**Comment by DeityLink** [comment]
Technically DeityLink but I was using this alt (JMWTurner) at the time for reasons.

**Comment by MistakeNot4892** [comment]
Apparently this is actually a Neb port of the Bay port of the Virgo port of the Aurora port of the vg implementation.

**Comment by DeityLink** [comment]
SS13 features do be like that. I think my explosion method and parallax also hopped through a few codebases.

File: code/controllers/subsystems/holomap.dm [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+				if (A.area_flags & AREA_FLAG_HIDE_FROM_HOLOMAP)
+					continue
+				if(IS_ROCK(tile))
+					continue
+				if(IS_OBSTACLE(tile))
+					canvas.DrawBox(HOLOMAP_OBSTACLE, x + offset_x, y + offset_y)
+				else if(IS_PATH(tile))
+					canvas.DrawBox(HOLOMAP_PATH, x + offset_x, y + offset_y)
+
+		CHECK_TICK
```
Comment (Kapu1178) [inline code/controllers/subsystems/holomap.dm]: y not just `block()`

File: code/modules/fabrication/designs/imprinter/designs_misc_circuits.dm [hunk 2]
```diff
@@ -477,4 +477,7 @@
 	path = /obj/item/stock_parts/circuitboard/inertial_damper
 
 /datum/fabricator_recipe/imprinter/circuit/docking_beacon
-	path = /obj/item/stock_parts/circuitboard/docking_beacon
\ No newline at end of file
+	path = /obj/item/stock_parts/circuitboard/docking_beacon
+
+/datum/fabricator_recipe/imprinter/circuit/docking_beacon
```
Comment (comma) [inline code/modules/fabrication/designs/imprinter/designs_misc_circuits.dm]: > /datum/fabricator_recipe/imprinter/circuit/docking_beacon

**Comment by MistakeNot4892** [comment]
Appears to be functional:
image

Would be nice if smaller maps got centered, but not critical for this PR.

**Comment by MistakeNot4892** [comment]
Never mind, something is cooked: 
image

**Comment by MistakeNot4892** [comment]
If anyone has any insight on why all the text is blue, it'd be appreciated.

**Comment by MistakeNot4892** [comment]
Oh for god's sake. I clicked 'close with comment' instead of deleting the comment (it was due to being a href) and since I was force pushing it killed the PR.

**Pull request closed without merging**